tests/rptest/services/redpanda.py

```diff
@@ -3239,6 +3239,14 @@ def write_bootstrap_cluster_config(self):
             conf.update(
                 dict(http_authentication=self._security.http_authentication))
 
+        # in redpanda, cloud_storage_spillover_manifest_size takes preference over cloud_storage_spillover_manifest_max_segments.
+        # disable the former to use the latter
+        assert conf.get('cloud_storage_spillover_manifest_size', None) is None or \
+            conf.get('cloud_storage_spillover_manifest_max_segments', None) is None, \
+                  "cannot set cloud_storage_spillover_manifest_max_segments if cloud_storage_spillover_manifest_size is already set, it will not be used by redpanda"
+        if 'cloud_storage_spillover_manifest_max_segments' in conf:
+            conf['cloud_storage_spillover_manifest_size'] = None
```

Comment:
This is surprising, and is something I can imagine us tripping over in the future.

Why isn't it sufficient to just set `cloud_storage_spillover_manifest_size` to None in the tests that set `..max_segments`?

Reply:
Some tests set the value in extra_rp_conf, some other set it in SISettings, i thought to just fix it in one place

Reply:
some other related tests are failing on the backport, I'll fix what there but just skip the redpanda build in the ci

Reply:
I'm also curious about this. How many tests does this affect can we just update a couple tests explicitly?

Reply:
Got it. It is a little annoying, but I think it's preferable since it's less surprising than having this business logic encoded in the service itself. For instance if we wanted to test these two configs together, which isn't an unreasonable thing for an unsuspecting user to do, we wouldn't be able to now